src/api/trips_v2.ts

```diff
@@ -22,14 +22,36 @@ export async function tripsSearch(
     when: query.when,
   };
 
-  const data = await post<TripsQuery>(url, cleanQuery, opts);
+  let data = await post<TripsQuery>(url, cleanQuery, opts);
+
+  if (data.trip?.tripPatterns) {
+    data.trip.tripPatterns = data.trip?.tripPatterns.map((tripPattern) => {
+      (tripPattern as TripPattern).id = generateIdFromTripPattern(tripPattern);
```

Comment:
Hvis den id'en kun skal brukes til `key` for React-elementer så synes denne bør genereres nært der den brukes (ikke i API-integrasjonskode). Det er samtidig misvisende å kalle dette id, siden den ikke kan brukes mot oppslag til BFF eller deep-link (som Mikael nevner her). Jeg tror vi bør etterstrebe å finne et pattern som nøster opp det nærliggende problemet (unik nøkkel for å liste ut React-elementer), samtidig som det enabler deep-linking, samt deling og lagring av reiseforslag o.l.

Reply:
høres ut som vi må snakke sammen

Reply:
Jeg ville kanskje ikke modifisert selve objekt med en id, da går vinningen litt opp i spinningen med denne endringen hvor man ikke ville ha en "hjemmelaget" id som ikke var en faktisk id fra Entur. Kunne man ikke gjort dette nært der hvor man bruker det som React key?

Reply:
jo, enig. Jeg og @cbrevik snakket også om det. rendyrke det som en react-key og ikke tilby det som en id.
Deeplinking, deling og offline lagring får nesten bli en sak for en annen gang, og det er flere hensyn som må tas her.
For nå så løses problemed med react key og oppdatering av reiseforslag. (etter flytting av key generation inn til komponenten som viser forslagene)